### PR TITLE
feat(kit): `InputDate` add full support of `TUI_DATE_FORMAT`

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes # 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/utils/mask/create-auto-corrected-date-pipe.ts [hunk 1]
```diff
@@ -1,17 +1,24 @@
-import {DATE_FILLER_LENGTH, TuiDay} from '@taiga-ui/cdk';
+import {DATE_FILLER_LENGTH, TuiDateMode, TuiDay} from '@taiga-ui/cdk';
 import {TuiTextMaskPipeHandler, TuiWithOptionalMinMaxWithValue} from '@taiga-ui/core';
 
+export interface AutoCorrectedDatePipeConfigs
```
Comment (splincode) [inline projects/kit/utils/mask/create-auto-corrected-date-pipe.ts:4]: why not `Tui` prefix?

File: projects/kit/components/input-date-time/input-date-time.component.ts [hunk 2]
```diff
@@ -314,10 +316,23 @@ export class TuiInputDateTimeComponent
...
         return {
-            mask: [...TUI_DATE_MASK, ',', ' ', ...tuiCreateTimeMask(timeMode)],
-            pipe: tuiCreateAutoCorrectedDateTimePipe({value: day, min, max}, timeMode),
+            // TODO finish localization in {@link https://github.com/TinkoffCreditSystems/taiga-ui/issues/954 issue}
+            mask: [
+                ...tuiCreateDateMask('DMY', '.'),
+                ',',
+                ' ',
+                ...tuiCreateTimeMask(timeMode),
+            ],
```
Comment (splincode) [inline projects/kit/components/input-date-time/input-date-time.component.ts:328]: What do you think about? We could move this code elsewhere. And indicate that in the future the internal implementation will change.

```ts
mask: tuiInternalCreateMask(timeMode)
```

```ts
/**
 * @note: TODO finish localization in {@link https://github.com/TinkoffCreditSystems/taiga-ui/issues/954 issue}
**/
export function tuiInternalCreateMask(timeMode) {
  return [
      ...tuiCreateDateMask('DMY', '.'),
      ',',
      ' ',
      ...tuiCreateTimeMask(timeMode),
   ];
}
```

File: projects/kit/components/input-date-time/input-date-time.component.ts [hunk 3]
```diff
@@ -314,10 +316,23 @@ export class TuiInputDateTimeComponent
         min: TuiDay,
         max: TuiDay,
         timeMode: TuiTimeMode,
+        dateFormat: TuiDateMode,
     ): TuiTextMaskOptions {
         return {
-            mask: [...TUI_DATE_MASK, ',', ' ', ...tuiCreateTimeMask(timeMode)],
-            pipe: tuiCreateAutoCorrectedDateTimePipe({value: day, min, max}, timeMode),
+            // TODO finish localization in {@link https://github.com/TinkoffCreditSystems/taiga-ui/issues/954 issue}
+            mask: [
```
Comment (splincode) [inline projects/kit/components/input-date-time/input-date-time.component.ts:323]: why can't we use `tuiCreateDateMask` here?

Comment (nsbarsukov) [reply projects/kit/components/input-date-time/input-date-time.component.ts:323]: sorry, i dont understand your question((

`tuiCreateDateMask` is already used here

File: projects/kit/utils/mask/create-auto-corrected-date-pipe.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/kit/utils/mask/create-auto-corrected-date-pipe.ts:4]: i will add this prefix, thanks

File: projects/kit/components/input-date-time/input-date-time.component.ts (hunk 3 above)
Comment (splincode) [reply projects/kit/components/input-date-time/input-date-time.component.ts:323]: Oh, sorry, I fault (( I read wrong

**Comment by nsbarsukov** [comment]
This PR is a part of https://github.com/TinkoffCreditSystems/taiga-ui/issues/954

File: projects/kit/utils/mask/create-date-mask.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import {tuiAssert, TuiDateMode} from '@taiga-ui/cdk';
+import {TUI_DIGIT_REGEXP, TuiTextMaskList} from '@taiga-ui/core';
+
+export function tuiCreateDateMask(mode: TuiDateMode, separator: string): TuiTextMaskList {
```
Comment (splincode) [inline projects/kit/utils/mask/create-date-mask.ts:4]: Could you write a separate test exclusively for this util? I would appreciate.

```ts
describe('tuiCreateDateMask', () => {
  it('YMD', () => {
    // ..
  });

  it('MDY', () => {
    // ..
  });

  it('DMY', () => {
    // ..
  });
})
```

**Review by splincode — APPROVED** [review]

File: projects/kit/utils/mask/create-date-mask.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    tuiAssert.assert(separator.length === 1, 'Separator should consist of only 1 symbol');
+
+    const DAY = [TUI_DIGIT_REGEXP, TUI_DIGIT_REGEXP];
```
Comment (waterplea) [inline projects/kit/utils/mask/create-date-mask.ts:7]: nit: `Array(2).fill(TUI_DIGIT_REGEXP)`. Do you think we also need both `DAY` and `MONTH`? You can move them outside this function, btw. Also we use caps for constants outside function scopes anyway :)

File: projects/kit/utils/mask/create-date-mask.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    const MONTH = [TUI_DIGIT_REGEXP, TUI_DIGIT_REGEXP];
+    const YEAR = [TUI_DIGIT_REGEXP, TUI_DIGIT_REGEXP, TUI_DIGIT_REGEXP, TUI_DIGIT_REGEXP];
+
+    switch (mode) {
+        case 'YMD':
+            return [...YEAR, separator, ...MONTH, separator, ...DAY];
+        case 'MDY':
+            return [...MONTH, separator, ...DAY, separator, ...YEAR];
+        default:
```
Comment (waterplea) [inline projects/kit/utils/mask/create-date-mask.ts:16]: hit: `default` not being the last option looks wrong to the eye :)

**Review by waterplea — APPROVED** [review]

**Pull request merged**